qcodes/instrument/channel.py

```diff
@@ -279,7 +264,7 @@ def index(self, obj):
         Args:
             obj(chan_type): The object to find in the channel list.
         """
-        return self._channels.index(obj)
+        return self._channels.index(obj) + self._oneindexed
```

Comment:
Maybe it's just me, but I would find `self._channels.index(obj) + int(self._oneindexed)` to be more obvious.